### PR TITLE
podmanv2 exit code

### PR DESCRIPTION
**Pull request by baude**
add ability to set and get exit code.

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: cmd/podmanV2/root.go [hunk 1]
```diff
@@ -30,7 +31,14 @@ func init() {
...
-		fmt.Println(err)
-		os.Exit(1)
+		fmt.Fprintln(os.Stderr, "Error:", err.Error())
+	} else if registry.GetExitCode() == define.ExecErrorCodeGeneric {
+		// The exitCode modified from define.ExecErrorCodeGeneric,
+		// indicates an application
+		// running inside of a container failed, as opposed to the
+		// podman command failed.  Must exit with that exit code
+		// otherwise command exited correctly.
+		registry.SetExitCode(0)
```
Comment (rhatdan) [inline cmd/podmanV2/root.go]: Why would you set the exit code to 0?

Comment (baude) [reply cmd/podmanV2/root.go]: i didnt ....
```67165b7675 cmd/podman/main.go (Valentin Rothberg 2020-01-13 13:01:45 +0100 170)                 exitCode = 0
```

Comment (rhatdan) [reply cmd/podmanV2/root.go]: @vrothberg PTAL

Comment (vrothberg) [reply cmd/podmanV2/root.go]: Commit 67165b7675 did _not_ set that but it removed a redundant check (tautology) to make a linter happy. Digging a bit deeper in the git history brings us to a Mr. Daniel J Walsh with commit b231e3412e7d :smile:

Comment (vrothberg) [reply cmd/podmanV2/root.go]: As far as I read it, we reserved `125` for for podman failures (see podman(1)). So this check seems like a safety net to make sure we don't return `125` when podman behaved correctly but the container might return `125`. In that case, we're losing the container's exit code.

Comment (rhatdan) [reply cmd/podmanV2/root.go]: Ok, I hope that Daniel J Walsh was correct...

**Comment by rhatdan** [comment]
LGTM
@vrothberg @mheon @TomSweeneyRedHat @jwhonce @QiWang19 PTAL

File: cmd/podmanV2/root.go (hunk 1 above)
Comment (vrothberg) [reply cmd/podmanV2/root.go]: He usually is :^)

**Pull request merged**